### PR TITLE
vlang: 0.1.21 -> weekly.2021.25

### PR DESCRIPTION
**Pull request by nagy**
###### Motivation for this change
Addresses #128063 
Supersedes #74985

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/vlang/default.nix [hunk 1]
```diff
@@ -4,32 +4,33 @@ assert stdenv.hostPlatform.isUnix -> upx != null;
 
 stdenv.mkDerivation rec {
   pname = "vlang";
-  version = "0.1.21";
+  version = "unstable-2021-06-21";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/vlang/default.nix:7]:
```suggestion
  version = "weekly.2021.25";
```

File: pkgs/development/compilers/vlang/default.nix [hunk 2]
```diff
@@ -4,32 +4,33 @@ assert stdenv.hostPlatform.isUnix -> upx != null;
...
 
   src = fetchFromGitHub {
     owner = "vlang";
     repo = "v";
-    rev = version;
-    sha256 = "0npd7a7nhd6r9mr99naib9scqk30209hz18nxif27284ckjbl4fk";
+    rev = "weekly.2021.25";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/vlang/default.nix:12]:
```suggestion
    rev = version;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by remimimimi** [comment]
Gives build error
image

**Comment by ethancedwards8** [comment]
> I try to build package locally and it gives build error @nagy
> image

Please post text logs, as images make it harder for everyone. Also text posts are more accessible to people with screen readers for example.

**Comment by SuperSandro2000** [comment]
> I try to build package locally and it gives build error @nagy

Can't reproduce on amd64.

**Comment by nagy** [comment]
> I try to build package locally and it gives build error @nagy

Yes I saw that as well. But it is just an error in one of the shipped examples. The build succeeds.

**Pull request merged**